Feature/visualising architecture (#11)
madetech/architecture-handbook: learning_paths/visualising_architecture/2_c4_model.md

Reviewer: Is there a `1_c4_model.md`? Can we remove the prefix?

Author: The prefix is to give the files within this folder an order based on how they're intended to be read

I was thinking of doing the same for the other path, as the pages do link together in a linear flow

# C4 Model

> The [C4 Model](https://c4model.com/) focusses on the visual communication and documentation of software architecture. It doesn’t present a formalised, standardised method to communicate software architecture, it does provide a collection of lightweight ideas and techniques that we've observed to be useful.

Reviewer: I know focusses is a valid spelling but it always looks slightly odd. You can feel free to ignore this though.

```suggestion
The [C4 Model](https://c4model.com/) focuses on the visual communication and documentation of software architecture. It doesn’t present a formalised, standardised method to communicate software architecture, it does however provide a collection of lightweight ideas and techniques that we've observed to be useful.
```

We've observed the use of the C4 Model with our active deliveries and found that it:
- Provides a lean approach to diagramming a software architecture at multiple levels

> - It helps focus on providing clear and concise meaning

Reviewer: ```suggestion
- Helps focus on providing clear and concise meaning
```

To keep consistency with the rest to avoid repetition of the `it`

- Is useful as starting point for other techniques such as risk storming
- Has a wide range of diagram types that are easy to learn, create and understand
- Isn't mandatory to create versions of all of the diagrams, only use ones that appropriate. System Context and Container diagrams are the most widely used.

> - is a simple but not simplistic approach to visualising architecture

Reviewer: ```suggestion
- Is a simple but not simplistic approach to visualising architecture
```

> The creator of C4 Simon Brown has published a called `Visualising Software Architecture`, which is part of our [recommended reading list](/resources.md)

Reviewer: ```suggestion
The creator of C4 Simon Brown has published a book called `Visualising Software Architecture`, which is part of our [recommended reading list](/resources.md)
```

## Structurizr
We recommend using the [C4 Model](https://c4model.com/) in conjunction with the [Structurizr DSL](https://github.com/structurizr/dsl) as a way to automatically generate the C4 diagrams from code. This allows for the Structurizr code to be kept in the same repository as the code of the system, making it easier to keep the two up to date together.

 A [Structurizr quick start template](https://github.com/madetech/structurizr-template) has been created that you can use as a starting point in your own projects.

## Watch
### Visualising Architecture CoP session (22nd November 2022)

> [![Visualising Architecture](/images/visualising_architecture.jpg)](https://drive.google.com/file/d/14R3OpBbcN0xbU_XeQQfG7ocehkw_U0w3/view)

Reviewer: Is it worth adding a note to indicate it's a Made Tech internal video?

```suggestion
[![Visualising Architecture](/images/visualising_architecture.jpg)](https://drive.google.com/file/d/14R3OpBbcN0xbU_XeQQfG7ocehkw_U0w3/view) (Made Tech internal video)
```

> Stuart McKee demostrates how to create a common understanding and foster collaboration

Reviewer: ```suggestion
Stuart McKee demonstrates how to create a common understanding and foster collaboration
```
